components/core/CMakeLists.txt

```diff
@@ -105,37 +105,72 @@ else()
     message(FATAL_ERROR "Could not find ${CLP_LIBS_STRING} libraries for Boost")
 endif()
 
-# Find and setup fmt
-# NOTE:
-# - We only try to link to the static library
-# - spdlog uses fmt, so their versions need to be kept in-sync
-find_package(fmt 8.0.1 REQUIRED)
-if(fmt_FOUND)
-    message(STATUS "Found fmt ${fmt_VERSION}")
-else()
-    message(FATAL_ERROR "Could not find static libraries for fmt")
-endif()
+# Find and setup spdlog and fmt transitive dependency
 
-# Find and setup spdlog
+# Specify the minimum required spdlog version (can be overridden by the user)
+set (spdlog_MIN_VERSION "1.9.0" CACHE STRING "Minimum required version of spdlog")
 if(CLP_USE_STATIC_LIBS)
     # NOTE: On some Linux distributions (e.g. Ubuntu), the spdlog package only contains a dynamic
     # library. If the `find_package(spdlog)` call below fails, re-run
     # `tools/scripts/lib_install/<dist_name>/install-packages-from-source.sh` to build spdlog from
     # source.
     set(spdlog_USE_STATIC_LIBS ON)
 endif()
-find_package(spdlog 1.9.2 REQUIRED)
+find_package(spdlog ${spdlog_MIN_VERSION} REQUIRED)
 if(spdlog_FOUND)
     message(STATUS "Found spdlog ${spdlog_VERSION}")
 else()
+    message("You may want to re-run `components/core/tools/scripts/lib_install/<dist>/install-packages-from-source.sh`")
     if (CLP_USE_STATIC_LIBS)
-        message(FATAL_ERROR "Could not find static libraries for spdlog. You may want to re-run
-            `components/core/tools/scripts/lib_install/<dist>/install-packages-from-source.sh`")
+        message(FATAL_ERROR "Could not find static libraries for spdlog.")
     else()
         message(FATAL_ERROR "Could not find libraries for spdlog.")
     endif()
 endif()
```

Comment:
Because `REQUIRED` causes a FATAL error if the package isn't found, the `else` block won't be executed. We should clean up this CMakeLists.txt in another PR or wait for Bill's refactoring to take effect.